### PR TITLE
Search by date of birth and by email

### PR DESCRIPTION
**Pull request by elahti**

### CODE REVIEW

File: test/cljs/unit/ataru/dob_test.cljs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  (:require [ataru.dob :as dob]
+            [cljs.test :refer-macros [deftest is testing]]))
+
+(deftest dob-identification
+  (doall
+    (for [[dob expected] [["1.1.2010" true]
+                          ["01.01.2010" true]
+                          ["1.01.2010" true]
+                          ["01.1.2010" true]
+                          ["1.13.2010" true] ; purposefully true
```
Comment (starvinraven) [inline test/cljs/unit/ataru/dob_test.cljs]: But why?

Comment (elahti) [inline test/cljs/unit/ataru/dob_test.cljs]: Improved the comment there. Problem is that `cljs.time` library automatically converts `1.13.2010` to `1.1.2011` when parsing the date. This means that invalid dates like `1.13.2010` or `40.10.2010` would have to be parsed manually, outside cljs.time.

At least for I thought this is a bit too much effort to implement properly.

Please note that on the clj side, in `ataru.dob-spec`, the same namespace is tested with different expectations.

File: src/cljs/ataru/virkailija/application/application_search_control.cljs [hunk 2]
```diff
@@ -21,35 +21,34 @@
...
+       [:span.application__search-control-clear-search-term
         {:on-click #(dispatch [:application/clear-applications-haku-and-form-selections])}
         [:i.zmdi.zmdi-close]])]))
 
-(defn search-ssn-tab [tab-id selected-tab link-url label-text]
+(defn search-term-tab [tab-id selected-tab link-url label-text]
   (let [tab-selected (when (= tab-id selected-tab) "application__search-control-selected-tab-with-input")]
     [:div.application__search-control-tab-selector-wrapper
      [:a {:href link-url
           :on-click (fn [event]
```
Comment (starvinraven) [inline src/cljs/ataru/virkailija/application/application_search_control.cljs]: This click-handler is called even when the search field is already visible. In other words, doing the following

1. Clicking on the "Etsi henkilötunnuksella..." button
2. Inputting e.g. a birthdate "1.1.2000" to the field which appear in place of the button
3. Clicking on the search results (or in any way blurring the input field)
4. Clicking back on the input field

.. resets the search field and also clears any search results. The reason seems to be that the button click handler (from step 1) is triggered even though the search field is already visible (which triggers the `/applications/search/` front route, resetting the state).

Comment (elahti) [inline src/cljs/ataru/virkailija/application/application_search_control.cljs]: Fixed.

**Pull request merged**